src/human_encoding/named_node.rs

```diff
@@ -124,23 +124,26 @@ impl<J: Jet> NamedCommitNode<J> {
         &self,
         witness: &HashMap<Arc<str>, Arc<Value>>,
         disconnect: &HashMap<Arc<str>, Arc<NamedCommitNode<J>>>,
-    ) -> Arc<WitnessNode<J>> {
+    ) -> Result<Arc<WitnessNode<J>>, crate::Error> {
         struct Populator<'a, J: Jet>(
             &'a HashMap<Arc<str>, Arc<Value>>,
             &'a HashMap<Arc<str>, Arc<NamedCommitNode<J>>>,
             PhantomData<J>,
         );
 
         impl<'a, J: Jet> Converter<Named<Commit<J>>, Witness<J>> for Populator<'a, J> {
-            type Error = ();
+            type Error = crate::Error;
 
             fn convert_witness(
                 &mut self,
                 data: &PostOrderIterItem<&Node<Named<Commit<J>>>>,
                 _: &NoWitness,
             ) -> Result<Option<Arc<Value>>, Self::Error> {
                 let name = &data.node.cached_data().name;
-                let value = self.0.get(name).ok_or(())?;
+                let value = self
```

Comment:
I guess for conditionals, even if we are not using the branch, we expect the user to specify some value of the correct type?

Reply:
Good catch. I changed the method to keep empty the witness nodes that don't have a mapping. There will be an error during finalization if an unpruned node is empty. Similar for disconnected branches / holes.